File: ios/mobile/BaseHaptic/Components/WatchInstallCard.swift
import SwiftUI

struct WatchInstallCard: View {
    let status: WatchCompanionStatus
    let onInstall: () -> Void
    let onRecheck: () -> Void
    let onOpenWatchApp: () -> Void
    let onWatchTest: () -> Void

    @Environment(\.teamTheme) private var teamTheme
    @State private var showFallbackAlert = false
    @State private var showInstructions = false

    private static let successGreen = Color(red: 76/255, green: 175/255, blue: 80/255)
    private static let warnRed = Color(red: 229/255, green: 57/255, blue: 53/255)

    var body: some View {
        Group {
            switch status {
            case .installed:
                stateBody(
                    glowColor: Self.successGreen,
                    badgeColor: Self.successGreen,
                    badgeIcon: "checkmark",
                    title: "워치 앱 설치됨",
                    description: "[워치 테스트]에서 체험해보세요",
                    primary: ActionSpec(
                        label: "워치 테스트",
                        sfSymbol: "applewatch.radiowaves.left.and.right",
                        action: onWatchTest,
                        enabled: true,
                        color: teamTheme.primary
                    ),
                    secondary: nil
                )
            case .pairedNoApp:
                stateBody(
                    glowColor: Self.warnRed,
                    badgeColor: Self.warnRed,
                    badgeIcon: "exclamationmark",
                    title: "워치 앱 설치 필요",
                    description: "Apple Watch앱에서 야구봄을 설치해 주세요.\n자동 설치가 켜져 있다면 잠시 후 자동으로 설치됩니다.",
                    primary: ActionSpec(
                        label: "연결 방법 상세 보기",
                        sfSymbol: "info.circle.fill",
                        action: { showInstructions = true },
                        enabled: true,
                        color: teamTheme.primary
                    ),
                    secondary: nil
                )
            case .pairedNone:
                stateBody(
                    glowColor: AppColors.gray500,
                    badgeColor: AppColors.gray700,
                    badgeIcon: "exclamationmark",
                    title: "워치 페어링 필요",
                    description: "iPhone에 Apple Watch를 먼저 페어링해 주세요.",
                    primary: ActionSpec(
                        label: "Watch 앱 열기",
                        sfSymbol: "arrow.up.right.square.fill",
                        action: onOpenWatchApp,
                        enabled: true,
                        color: teamTheme.primary
                    ),
                    secondary: ActionSpec(
                        label: "연결 확인",
                        sfSymbol: "link",
                        action: onRecheck,
                        enabled: true,
                        color: .clear
                    )
                )
            case .loading:
                compactLoadingRow
            }
        }
        .background(AppColors.gray900)
        .cornerRadius(AppRadius.md)
        .alert("설치 안내", isPresented: $showFallbackAlert) {
            Button("확인", role: .cancel) {}
        } message: {
            Text("iPhone의 Apple Watch 앱을 직접 열어 사용 가능한 앱 목록에서 야구봄을 설치해 주세요.")
        }
        .sheet(isPresented: $showInstructions) {
            WatchInstallInstructionsSheet(onDismiss: { showInstructions = false })
        }
    }

    private struct ActionSpec {
        let label: String
        let sfSymbol: String
        let action: () -> Void
        let enabled: Bool
        let color: Color
    }

    private var compactLoadingRow: some View {
        HStack(spacing: AppSpacing.md) {
            Image(systemName: "applewatch")
                .foregroundColor(AppColors.gray400)
                .font(AppFont.h4)
            Text("워치 상태 확인 중…")
                .font(AppFont.bodyLgMedium)
                .foregroundColor(AppColors.gray400)
            Spacer()
        }
        .padding(AppSpacing.lg)
    }

    private func stateBody(
        glowColor: Color,
        badgeColor: Color,
        badgeIcon: String,
        title: String,
        description: String,
        primary: ActionSpec?,
        secondary: ActionSpec?
    ) -> some View {
        VStack(spacing: 0) {
            HStack {
                Text("워치 앱")
                    .font(AppFont.bodyMedium)
                    .foregroundColor(AppColors.gray400)
                Spacer()
            }
            .padding(.bottom, AppSpacing.md)

            watchIllustration(glowColor: glowColor, badgeColor: badgeColor, badgeIcon: badgeIcon)

            Text(title)
                .font(AppFont.bodyLgBold)
                .foregroundColor(.white)
                .padding(.top, AppSpacing.md)

            Text(description)
                .font(AppFont.body)
                .foregroundColor(AppColors.gray400)
                .multilineTextAlignment(.center)
                .padding(.top, AppSpacing.xs)

            if let primary = primary {
                primaryButton(primary)
                    .padding(.top, AppSpacing.lg)
            }
            if let secondary = secondary {
                secondaryButton(secondary)
                    .padding(.top, AppSpacing.sm)
            }
        }
        .padding(AppSpacing.lg)
    }

    private func watchIllustration(glowColor: Color, badgeColor: Color, badgeIcon: String) -> some View {
        ZStack {
            // 상태색 radial glow — 워치 아이콘 뒤에서 부드럽게 퍼지는 효과
            Circle()
                .fill(
                    RadialGradient(
                        gradient: Gradient(colors: [
                            glowColor.opacity(0.55),
                            glowColor.opacity(0.0)
                        ]),
                        center: .center,
                        startRadius: 0,
                        endRadius: 70
                    )
                )
                .frame(width: 140, height: 140)
                .blur(radius: 8)

            ZStack {
                Circle()
                    .fill(Color.white.opacity(0.1))
                    .frame(width: 96, height: 96)
                Image(systemName: "applewatch")
                    .resizable()
                    .scaledToFit()
                    .frame(width: 56, height: 56)
                    .foregroundColor(.white)
            }

            ZStack(alignment: .topTrailing) {
                Color.clear
                    .frame(width: 120, height: 120)
                ZStack {
                    Circle()
                        .fill(badgeColor)
                        .frame(width: 28, height: 28)
                    Image(systemName: badgeIcon)
                        .foregroundColor(.white)
                        .font(.system(size: 14, weight: .bold))
                }
                .offset(x: 8, y: -8)
            }
        }
        .frame(width: 140, height: 140)
    }

    private func primaryButton(_ spec: ActionSpec) -> some View {
        Button(action: spec.action) {
            HStack(spacing: AppSpacing.sm) {
                Image(systemName: spec.sfSymbol)
                Text(spec.label)
                    .font(AppFont.bodyLgMedium)
            }
            .foregroundColor(spec.enabled ? .white : AppColors.gray400)
            .frame(maxWidth: .infinity)
            .frame(height: 48)
            .background(spec.enabled ? spec.color : AppColors.gray800)
            .cornerRadius(AppRadius.sm)
        }
        .disabled(!spec.enabled)
    }

    private func secondaryButton(_ spec: ActionSpec) -> some View {
        Button(action: spec.action) {
            HStack(spacing: AppSpacing.sm) {
                Image(systemName: spec.sfSymbol)
                Text(spec.label)
                    .font(AppFont.body)
            }
            .foregroundColor(AppColors.gray300)
            .frame(maxWidth: .infinity)
            .frame(height: 44)
            .overlay(
                RoundedRectangle(cornerRadius: AppRadius.sm)
                    .stroke(AppColors.gray700, lineWidth: 1)
            )
        }
    }
}

File: apps/mobile/app/src/main/java/com/basehaptic/mobile/ui/components/WatchInstallCard.kt
package com.basehaptic.mobile.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Download
import androidx.compose.material.icons.filled.Link
import androidx.compose.material.icons.automirrored.filled.OpenInNew
import androidx.compose.material.icons.filled.PriorityHigh
import androidx.compose.material.icons.filled.Watch
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.basehaptic.mobile.ui.theme.AppFont
import com.basehaptic.mobile.ui.theme.AppShapes
import com.basehaptic.mobile.ui.theme.AppSpacing
import com.basehaptic.mobile.ui.theme.Gray300
import com.basehaptic.mobile.ui.theme.Gray400
import com.basehaptic.mobile.ui.theme.Gray500
import com.basehaptic.mobile.ui.theme.Gray700
import com.basehaptic.mobile.ui.theme.Gray800
import com.basehaptic.mobile.ui.theme.Gray900
import com.basehaptic.mobile.ui.theme.LocalTeamTheme
import com.basehaptic.mobile.wear.WatchCompanionStatus

private val SuccessGreen = Color(0xFF4CAF50)
private val WarnRed = Color(0xFFE53935)

@Composable
fun WatchInstallCard(
    status: WatchCompanionStatus,
    onInstall: () -> Unit,
    onRecheck: () -> Unit,
    onOpenWatchApp: () -> Unit,
    onWatchTest: () -> Unit,
    modifier: Modifier = Modifier
) {
    val theme = LocalTeamTheme.current
    Surface(
        modifier = modifier.fillMaxWidth(),
        shape = AppShapes.md,
        color = Gray900,
        tonalElevation = 1.dp
    ) {
        when (status) {
            is WatchCompanionStatus.Installed -> StateBody(
                glowColor = SuccessGreen,
                badgeColor = SuccessGreen,
                badgeIcon = Icons.Default.Check,
                title = "워치 앱 설치됨",
                description = "[워치 테스트]에서 체험해보세요",
                primary = ActionSpec(
                    label = "워치 테스트",
                    icon = Icons.Default.Watch,
                    onClick = onWatchTest,
                    enabled = true,
                    color = theme.primary
                ),
                secondary = null
            )
            is WatchCompanionStatus.PairedNoApp -> StateBody(
                glowColor = WarnRed,
                badgeColor = WarnRed,
                badgeIcon = Icons.Default.PriorityHigh,
                title = "워치 앱 설치 필요",
                description = "Play Store에서 설치 대상 기기를 워치로 선택해 주세요.",
                primary = ActionSpec(
                    label = "워치 앱 설치하기",
                    icon = Icons.Default.Download,
                    onClick = onInstall,
                    enabled = true,
                    color = theme.primary
                ),
                secondary = ActionSpec(
                    label = "연결 확인",
                    icon = Icons.Default.Link,
                    onClick = onRecheck,
                    enabled = true,
                    color = Color.Transparent
                )
            )
            // PairedNone 케이스
            is WatchCompanionStatus.PairedNone -> StateBody(
                glowColor = Gray500,
                badgeColor = Gray700,
                badgeIcon = Icons.Default.PriorityHigh,
                title = "워치 페어링 필요",
                description = "Wear OS 앱에서 워치를 폰과 먼저 페어링해 주세요.",
                primary = ActionSpec(
                    label = "Wear OS 앱 열기",
                    icon = Icons.AutoMirrored.Filled.OpenInNew,
                    onClick = onOpenWatchApp,
                    enabled = true,
                    color = theme.primary
                ),
                secondary = ActionSpec(
                    label = "연결 확인",
                    icon = Icons.Default.Link,
                    onClick = onRecheck,
                    enabled = true,
                    color = Color.Transparent
                )
            )
            is WatchCompanionStatus.Loading -> CompactLoadingRow()
        }
    }
}

private data class ActionSpec(
    val label: String,
    val icon: ImageVector,
    val onClick: () -> Unit,
    val enabled: Boolean,
    val color: Color
)

@Composable
private fun StateBody(
    glowColor: Color,
    badgeColor: Color,
    badgeIcon: ImageVector,
    title: String,
    description: String,
    primary: ActionSpec?,
    secondary: ActionSpec?
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(AppSpacing.lg),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Row(modifier = Modifier.fillMaxWidth()) {
            Text(
                text = "워치 앱",
                style = AppFont.bodyMedium,
                color = Gray400
            )
        }
        Spacer(modifier = Modifier.height(AppSpacing.md))

        WatchIllustration(glowColor = glowColor, badgeColor = badgeColor, badgeIcon = badgeIcon)

        Spacer(modifier = Modifier.height(AppSpacing.md))

        Text(
            text = title,
            style = AppFont.bodyLgBold,
            color = Color.White
        )
        Spacer(modifier = Modifier.height(AppSpacing.xs))
        Text(
            text = description,
            style = AppFont.body,
            color = Gray400,
            textAlign = TextAlign.Center
        )

        if (primary != null) {
            Spacer(modifier = Modifier.height(AppSpacing.lg))
            PrimaryButton(spec = primary)
        }
        if (secondary != null) {
            Spacer(modifier = Modifier.height(AppSpacing.sm))
            SecondaryButton(spec = secondary)
        }
    }
}

@Composable
private fun WatchIllustration(
    glowColor: Color,
    badgeColor: Color,
    badgeIcon: ImageVector
) {
    Box(
        modifier = Modifier.size(120.dp),
        contentAlignment = Alignment.Center
    ) {
        // Reason: 상태색 radial glow — 워치 아이콘 뒤에서 부드럽게 퍼지는 효과
        Box(
            modifier = Modifier
                .size(110.dp)
                .blur(28.dp)
                .background(
                    brush = Brush.radialGradient(
                        colors = listOf(
                            glowColor.copy(alpha = 0.55f),
                            glowColor.copy(alpha = 0.0f)
                        )
                    ),
                    shape = CircleShape
                )
        )
        Box(
            modifier = Modifier
                .size(96.dp)
                .background(Color(0x1AFFFFFF), CircleShape),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = Icons.Default.Watch,
                contentDescription = null,
                tint = Color.White,
                modifier = Modifier.size(56.dp)
            )
        }
        Box(
            modifier = Modifier
                .size(120.dp),
            contentAlignment = Alignment.TopEnd
        ) {
            Box(
                modifier = Modifier
                    .size(28.dp)
                    .background(badgeColor, CircleShape),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = badgeIcon,
                    contentDescription = null,
                    tint = Color.White,
                    modifier = Modifier.size(18.dp)
                )
            }
        }
    }
}

@Composable
private fun PrimaryButton(spec: ActionSpec) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .height(48.dp)
            .clip(AppShapes.sm)
            .clickable(enabled = spec.enabled, onClick = spec.onClick),
        shape = AppShapes.sm,
        color = if (spec.enabled) spec.color else Gray800
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                imageVector = spec.icon,
                contentDescription = null,
                tint = if (spec.enabled) Color.White else Gray400,
                modifier = Modifier.size(18.dp)
            )
            Spacer(modifier = Modifier.width(AppSpacing.sm))
            Text(
                text = spec.label,
                style = AppFont.bodyLgMedium,
                color = if (spec.enabled) Color.White else Gray400
            )
        }
    }
}

@Composable
private fun SecondaryButton(spec: ActionSpec) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .height(44.dp)
            .clip(AppShapes.sm)
            .clickable(onClick = spec.onClick),
        shape = AppShapes.sm,
        color = Color.Transparent,
        border = BorderStroke(1.dp, Gray700)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                imageVector = spec.icon,
                contentDescription = null,
                tint = Gray300,
                modifier = Modifier.size(16.dp)
            )
            Spacer(modifier = Modifier.width(AppSpacing.sm))
            Text(
                text = spec.label,
                style = AppFont.body,
                color = Gray300
            )
        }
    }
}

@Composable
private fun CompactLoadingRow() {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(AppSpacing.lg),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = Icons.Default.Watch,
            contentDescription = null,
            tint = Gray400,
            modifier = Modifier.size(24.dp)
        )
        Spacer(modifier = Modifier.width(AppSpacing.md))
        Text(
            text = "워치 상태 확인 중…",
            style = AppFont.bodyLgMedium,
            color = Gray400
        )
    }
}
